### PR TITLE
gecko: EFR32MG24 Device files updated | Update to GSDK 4.3.2

### PR DESCRIPTION
**Pull request by sateeshkotapati**
Origin: Silicon Labs Gecko SDK
URL: https://github.com/SiliconLabs/gecko_sdk
Version: 4.3.2
Purpose: Update hal_silabs to gecko_sdk v4.3.2.

Update the EFR32MG24 device files inside `gecko/Device/SiliconLab/EFR32MG24`  from gecko_sdk to align the codebase of hal_silabs with gecko_sdk 4.3.2.

### CODE REVIEW

File: gecko/Device/SiliconLabs/EFR32MG24/Source/GCC/efr32mg24.ld [hunk 1]
```diff
@@ -0,0 +1,292 @@
+/***************************************************************************//**
+ * @file     gcc_EFR32MG24.ld
```
Comment (fkokosinski) [inline gecko/Device/SiliconLabs/EFR32MG24/Source/GCC/efr32mg24.ld:2]: This file won't be used by Zephyr, right? Let's avoid adding files which will not be potentially used by Zephyr.

File: gecko/Device/SiliconLabs/EFR32MG24/Source/IAR/startup_efr32mg24.s [hunk 2]
```diff
@@ -0,0 +1,540 @@
+;/**************************************************************************//**
+; * @file     startup_efr32mg24.S
+; * @brief    CMSIS-Core(M) Device Startup File for
+; *           Device EFR32MG24
+; * @version  V1.0.0
+; * @date     08. April 2021
+; ******************************************************************************
+; * # License
```
Comment (fkokosinski) [inline gecko/Device/SiliconLabs/EFR32MG24/Source/IAR/startup_efr32mg24.s:8]: Ditto

File: gecko/Device/SiliconLabs/EFR32MG24/Source/startup_efr32mg24.c [hunk 3]
```diff
@@ -0,0 +1,406 @@
+/******************************************************************************
+* @file     startup_efr32mg24.c
+* @brief    CMSIS-Core(M) Device Startup File for
+*           Device EFR32MG24
+* @version  V2.1.0
+* @date     16. December 2020
+*******************************************************************************
+* # License
```
Comment (fkokosinski) [inline gecko/Device/SiliconLabs/EFR32MG24/Source/startup_efr32mg24.c:8]: Ditto

**Review by fkokosinski — CHANGES_REQUESTED** [review]

File: gecko/Device/SiliconLabs/EFR32MG24/Source/GCC/startup_efr32mg24.S [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+ *           Device EFR32MG24
+ * @version  V2.3.0
+ * @date     26. May 2021
+ ******************************************************************************
+ ******************************************************************************
+ * # License
+ *
+ * The licensor of this software is Silicon Laboratories Inc. Your use of this
+ * software is governed by the terms of Silicon Labs Master Software License
+ * Agreement (MSLA) available at
```
[the remark this answers is not in the document]
Comment (sateeshkotapati) [reply gecko/Device/SiliconLabs/EFR32MG24/Source/GCC/startup_efr32mg24.S]: Okay, I have removed the mentioned file as it is not necessary.

File: gecko/Device/SiliconLabs/EFR32MG24/Source/GCC/efr32mg24.ld (hunk 1 above)
Comment (sateeshkotapati) [reply gecko/Device/SiliconLabs/EFR32MG24/Source/GCC/efr32mg24.ld]: Removed the file

File: gecko/Device/SiliconLabs/EFR32MG24/Source/IAR/startup_efr32mg24.s (hunk 2 above)
Comment (sateeshkotapati) [reply gecko/Device/SiliconLabs/EFR32MG24/Source/IAR/startup_efr32mg24.s]: Removed the file

File: gecko/Device/SiliconLabs/EFR32MG24/Source/startup_efr32mg24.c (hunk 3 above)
Comment (sateeshkotapati) [reply gecko/Device/SiliconLabs/EFR32MG24/Source/startup_efr32mg24.c]: Removed the file

**Comment by sateeshkotapati** [comment]
@fkokosinski, I have done the requested changes, can you please review this again.

**Review by fkokosinski — APPROVED** [review]
LGTM

**Pull request merged**